lib/constructs/aws-serverless-s3-stack.ts

```diff
@@ -0,0 +1,115 @@
+
+        // apply life cycle rules for json files bucket
+        this.createS3BucketLifecycleRule(jsonFilesBucket, `${props.resourcePrefix}-${props.jsonFilesBucketName}-LifecycleRule`);
+
+        // assign the json files bucket as an input for the next stack
+        this.jsonFilesBucketName = jsonFilesBucket.bucketName;
+    }
+
+    /**
+     * Creates an S3 bucket with specified configurations.
+     *
+     * @param {Construct} scope - The scope in which to define this construct.
+     * @param {string} bucketName - The name of the S3 bucket to create.
+     * @param {cdk.RemovalPolicy} removalPolicy - The removal policy to apply to the bucket.
+     * @returns {s3.Bucket} The created S3 bucket.
+     * @throws {Error} If bucket creation fails due to invalid parameters or AWS API errors.
+     */
+    createS3Bucket(scope: Construct, bucketName: string, removalPolicy: cdk.RemovalPolicy): s3.Bucket {
+        return new s3.Bucket(scope, bucketName, {
+            bucketName: bucketName,
+            encryption: s3.BucketEncryption.S3_MANAGED,
+            blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+            publicReadAccess: false,
+            removalPolicy: removalPolicy,
+            autoDeleteObjects: removalPolicy === cdk.RemovalPolicy.DESTROY,
+            accessControl: s3.BucketAccessControl.BUCKET_OWNER_FULL_CONTROL,
+            versioned: true, // Enable versioning
+            enforceSSL: true,
+        });
+    }
+
+    /**
+     * Applies lifecycle rules to an S3 bucket for efficient object management.
+     *
+     * This method configures rules for aborting incomplete multipart uploads,
+     * transitioning objects to different storage classes, and expiring noncurrent versions.
+     *
+     * @param {s3.Bucket} s3Bucket - The S3 bucket to which the lifecycle rules will be applied.
+     * @param {string} id - A unique identifier for the lifecycle rule.
+     * @throws {Error} If the lifecycle rule cannot be applied due to invalid parameters or AWS API errors.
+     */
+    createS3BucketLifecycleRule(s3Bucket: s3.Bucket, id: string) {
+        s3Bucket.addLifecycleRule({
+            id: id,
+            abortIncompleteMultipartUploadAfter: cdk.Duration.days(7), // After 7 days, incomplete multipart uploads will be aborted
+            enabled: true,
+            noncurrentVersionExpiration: cdk.Duration.days(365),
+            noncurrentVersionTransitions: [
+                {
+                    storageClass: cdk.aws_s3.StorageClass.INFREQUENT_ACCESS, // Transition to Infrequent Access storage class after 30 days
+                    transitionAfter: cdk.Duration.days(30),
+                },
+            ],
+            transitions: [
+                {
+                    storageClass: cdk.aws_s3.StorageClass.INTELLIGENT_TIERING, // Transition to Intelligent Tiering storage class after 60 days
+                    transitionAfter: cdk.Duration.days(60),
```

Comment:
**Suggestion:** Use a constant for the lifecycle rule duration values to avoid magic numbers and improve code readability.
> 🌤️  **Medium Priority** 
 > [Category: **`Best Practice`**, Importance: **`7`**]
> **Reason:** Using constants for lifecycle rule duration values enhances code readability and maintainability by avoiding magic numbers.

```suggestion
            const ABORT_MULTIPART_UPLOAD_DAYS = 7;
            const NONCURRENT_VERSION_EXPIRATION_DAYS = 365;
            const INFREQUENT_ACCESS_TRANSITION_DAYS = 30;
            const INTELLIGENT_TIERING_TRANSITION_DAYS = 60;

            abortIncompleteMultipartUploadAfter: cdk.Duration.days(ABORT_MULTIPART_UPLOAD_DAYS), // After 7 days, incomplete multipart uploads will be aborted
            noncurrentVersionExpiration: cdk.Duration.days(NONCURRENT_VERSION_EXPIRATION_DAYS),
            noncurrentVersionTransitions: [
              {
                storageClass: cdk.aws_s3.StorageClass.INFREQUENT_ACCESS, // Transition to Infrequent Access storage class after 30 days
                transitionAfter: cdk.Duration.days(INFREQUENT_ACCESS_TRANSITION_DAYS),
              },
            ],
            transitions: [
              {
                storageClass: cdk.aws_s3.StorageClass.INTELLIGENT_TIERING, // Transition to Intelligent Tiering storage class after 60 days
                transitionAfter: cdk.Duration.days(INTELLIGENT_TIERING_TRANSITION_DAYS),
              },
            ],
```
- [ ]  **I have reviewed this code suggestion and addressed the relevant ones.**